src/backend/joanie/core/filters/client.py

```diff
@@ -101,14 +101,31 @@ def filter_product_type(self, queryset, _name, value):
 class ContractViewSetFilter(filters.FilterSet):
     """ContractFilter allows to filter this resource with a signature state."""
 
-    is_signed = filters.BooleanFilter(method="get_is_signed")
+    state = filters.ChoiceFilter(
+        method="filter_state", choices=enums.CONTRACT_STATE_FILTER_CHOICES
+    )
```

Comment:
What about using `signature_state` instead of `state` ?
```suggestion
    signature_state = filters.ChoiceFilter(
        method="filter_state", choices=enums.CONTRACT_STATE_FILTER_CHOICES
    )
```